File: Java/680.java
import java.util.Scanner;

/**
 *
 * 680 - La Carrera del Salmón - Golgotha's Melancholy - https://github.com/Golgotha-sMelancholy
 */
public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int casos = sc.nextInt();
        for (int i = 1; i <= casos; i++) {
            int contador = 1;
            int x = sc.nextInt();
            int y = sc.nextInt();
            int z = sc.nextInt();
            if (z >= y && y < x) {
                System.out.println("IMPOSIBLE");
            } else {
                int suma = 0;
                while (suma+y < x) {
                    suma += (y - z);
                    contador++;
                }
                System.out.println(contador);
            }

        }
    }

}

File: C/680.c
#include <stdio.h>
#include <stdlib.h>

// 680 - La Carrera del Salmón - Golgotha's Melancholy - https://github.com/Golgotha-sMelancholy
int main()
{
    int loop = 0;
    scanf("%d", &loop);
    int i;
    for (i = 0; i<loop; i++) {
        int counter = 1;
        int x = 0; int y = 0; int z = 0;
        scanf("%d", &x);
        scanf("%d", &y);
        scanf("%d", &z);
        if (z >= y && y<x) {
            printf("IMPOSIBLE\n");
        } else {
            int total = 0;
            while (total+y<x) {
                total += (y-z);
                counter++;
            }
            printf("%d\n", counter);
        }
    }
    return 0;
}
